networking.md

```diff
@@ -0,0 +1,88 @@
+# Filecoin Network
+
+The filecoin network is built using libp2p building blocks, as transports and protocols, as well as some additional filecoin specific protocols as outlined in [Network Protocols](./network-protocols.md).
+
+## Required Protocols
+
+Every full node must support the following libp2p protocols:
+
+ - [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) ([for data announcements](./data-propagation.md))
+ - [bitswap] ([for data exchange](./data-propagation.md))
+ - Filecoin specific Protocols:
+   - Hello Handshake
+   - StorageDeal
+   - BlockSync
+
+## Transports, Streams & Encryption
+
+Connections between nodes are end-to-end-encrypted and -authenticated, thus every node has a `peerId` associated with it.
+
+Filecoin protocols are run as multiple streams over a single connection. Only one connection should exist between two nodes at any time, multiple streams within that connections must be used to exchange different protocols. This is usually handled from within the libp2p stack if configured properly.
+
+This document doesn't make any assumptions on which transports and multiplexing features a particular node offers, however it must at least be able to respond and communicate through mplex channels via a secio transport.
+
+{{% notice info %}}
+Implementers are encouraged to offer and experiment with more protocols and transports, as long as the above mentioned requirements of ee2e, authentication and muxing are still met. In particular the authors encourage to look into TLS1.3 and QUIC as libp2p protocol/transport layers.
+{{% /notice %}}
+
+### Establishing connections
+
+When a filecoin node connects to the network, it may connect to multiple nodes at once. When a new node connects, it must be greeted with the `HelloMessage` through the hello-handshake protocol first. Only after that message has been sent can other streams be opened. If a node receives a `HelloMessage` with a GenesisBlock it doesn't support, it must immediately close the entire connection to that peer.
+
+If the node learns through that handshake about newer blocks it should request these immediately to sync up their chain.
+
+### Syncing
+
+Whenever a node learns about a new `BlockHead` it should attempt to import the block - let that be through the `HelloMessage` or through the [block pubsub protocol](data-propagation.md#block-propagation). For that it may fetches the ancestors of that block through bitswap until it is fully caught up. Importing in this context refers to the node confirming that the block is valid, as described in [Validation](./validation.md) and storing it locally. This mode is known as "syncing". During "syncing" the same node may not mine blocks.
```

Comment:
What about „caught up“ mode? Is that not a thing anymore?

Reply:
typo "it may fetche**es**".

Reply:
@dignifiedquire I never quite understood the idea of distinct "modes", it for sure isn't how we have implemented syncing in substrate. I picture this might be a useful architecture in implementing it in certain frameworks or context, but from a spec perspective a node is a node and there aren't really distinct "modes" in which it is but just certain requirements that is has to adhere by when doing certain things. The only thing I found was the there obviously shouldn't be any "mining" while importing. But that could easily be implemented with a lock rather than understanding it as different modes. And that is much easier than thinking about all the cases of transitions between the modes that could occur IMHO.

Or is there anything in particular what I missed about these modes and that we are missing in this version?

Reply:
Afaict there is not more to these modes, but @whyrusleeping might have some additional thoughts here, as he was the one who added them in the first place I believe.